README.md

```diff
@@ -1,10 +1,102 @@
 # kv-go
-Distributed in-memory (at least for now) KV storage written in Go
 
-#Motivation
-Just For Fun
-This is an educational project about how to develop scalable distributed systems.
-
-#Disclaimer
+# Disclaimer
 Use this project at your own risk.
 PRs are welcome!
+
+## Distributed in-memory (at least for now) KV storage written in Go
+
+## Project Overview
+
+### Objective
+
+To design and implement a distributed in-memory key-value storage system using the Go programming language. The system should be capable of handling high throughput, be fault-tolerant, and easily scalable. This is an educational project about how to develop scalable distributed systems. Just for fun.
+
+
+### Scope
+
+1. In-Memory Key-Value Storage
+2. Data Replication
+3. Data Sharding
+4. Fault Tolerance
+5. Consistency Guarantees
+6. API for CRUD operations
+7. Monitoring and Logging
+
+## System Architecture
+
+### Components
+
+1. **Node**: Basic unit of the storage system, responsible for storing in-memory data.
+2. **Cluster Manager**: Manages the nodes in the system.
+3. **Client Library**: Used for interacting with the storage system.
```

Comment:
What do you mean by that? Cli application, or SDK?

Reply:
HI @biosvs, we agreed on CLI for the first iteration - it will be easier to start with and use